core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityManager.scala

```diff
@@ -1,39 +1,56 @@
 import org.broadinstitute.dsde.rawls.entities.local.LocalEntityProvider
-import org.broadinstitute.dsde.rawls.model.{AttributeEntityReference, Entity, Workspace}
+import org.broadinstitute.dsde.rawls.model.{RawlsBillingProject, UserInfo, Workspace}
 
-import scala.concurrent.Future
 import scala.reflect.runtime.universe._
 
 /**
- * Requests for entity operations enter here. This is the interface that calling code should use; calling
- * code should not directly access entity providers.
- * This manager will inspect the request and route it to the appropriate provider.
+ * Here's the philosophy behind the important entity classes:
+ *
+ * EntityProvider:
+ *    these do the nuts-and-bolts work of connecting to a datasource and manipulating the entities therein.
+ *    EntityProvider authors should not have to worry too much about thread safety, multitenancy, concurrency, etc - so,
+ *    we create a new EntityProvider instance for each request.
+ *
+ *    Subclasses are:
+ *      - LocalEntityProvider: the default. Legacy Rawls/CloudSQL implementation.
+ *      - DataRepoEntityProvider: for working with Terra Data Repo snapshots.
+ *
+ * EntityProviderBuilder:
+ *    since we create many instances of EntityProvider, we want a factory pattern. These builders are responsible
+ *    for making the various EntityProvider instances. Builders should be singletons, and can be instantiated
+ *    once with config values or other arguments that the provider instances will need.
+ *
+ * EntityManager:
+ *    another singleton, the EntityManager is instantiated with the set of ProviderBuilders that this application
+ *    knows about. The manager is responsible for inspecting the inbound request, determining which builder should
+ *    be used to satisfy the request, and using that builder to create and return a provider instance.
+ *
  */
 class EntityManager(providerBuilders: Set[EntityProviderBuilder[_ <: EntityProvider]]) {
 
-  /** create the supplied entity inside the supplied workspace.
-   */
-  def createEntity(workspace: Workspace, entity: Entity): Future[Entity] = {
-    resolveProvider(workspace).createEntity(entity)
-  }
-
-  // soon: this interface will have multiple other methods, for example:
-  def getEntity(workspace: Workspace, entityRef: AttributeEntityReference) = ???
-
-  // internal method to figure out which entityprovider should be used for a given request
-  private def resolveProvider(workspace: Workspace): EntityProvider = {
-    // soon: inspect the workspace's registered snapshots plus request parameters,
-    // and route to the appropriate entity provider. This will likely require a method signature change
-    // to accept request details. For now, just find the builder for DefaultEntityProvider.
-    val defaultEntityProviderBuilder = providerBuilders.find(_.builds == typeTag[LocalEntityProvider])
+  // should we combine any of these arguments into a "EntityRequestArguments" case class,
+  // for encapsulation/abstraction?
```

Comment:
yes, I think so. There is a little uncertainty here and it will allow for some flexibility.